File: bench/py/24_dict_lookup.py
# repeated dictionary lookups by key
import sys
scale = int(sys.argv[1]) if len(sys.argv) > 1 else 1

M = 10000
d = {}
for i in range(M):
    d[i] = i

N = 1000000 * scale
s = 0

for i in range(N):
    s += d[i % M]
    s = s % 1000000007

print("result:", s)

File: bench/cpp/24_dict_lookup.cpp
/* SPDX-License-Identifier: BSD-2-Clause */
/* Faithful C++ of bench/my/24_dict_lookup.my: repeated dictionary lookups by
 * integer key (std::unordered_map, the C++ hashmap MyLang's dict wraps). */
#include "bench.h"
#include <unordered_map>

int main(int argc, char **argv)
{
    long scale = bench_scale(argc, argv);

    const long M = 10000;
    std::unordered_map<long, long> d;
    for (long i = 0; i < M; i++)
        d[i] = i;

    long N = 1000000L * scale;
    long s = 0;
    for (long i = 0; i < N; i++) {
        s += d[i % M];
        s = s % 1000000007L;
    }

    printf("result: %ld\n", s);
    return 0;
}
